android/capacitor/src/main/java/com/getcapacitor/plugin/LocalNotification.java

```diff
@@ -0,0 +1,199 @@
+   */
+  @PluginMethod()
+  public void schedule(PluginCall call) {
+    List<Notification> notifications = this.buildNotificationList(call);
+    if (notifications == null) {
+      return;
+    }
+    JSONArray ids = new JSONArray();
+    NotificationManagerCompat notificationManager = NotificationManagerCompat.from(this.getContext());
+    boolean notificationsEnabled = notificationManager.areNotificationsEnabled();
+    if (!notificationsEnabled) {
+      call.error("Notifications not enabled on this device");
+      return;
+    }
+    for (Notification notification : notifications) {
+      if (notification.getId() == null) {
+        call.error("Notification missing identifier");
+        return;
+      }
+
+      Intent intent = new Intent(getContext(), getActivity().getClass());
+      intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
+      PendingIntent pendingIntent = PendingIntent.getActivity(getContext(), 0, intent, 0);
+
+      NotificationCompat.Builder mBuilder = new NotificationCompat.Builder(this.getContext(), CHANNEL_ID)
+              .setContentTitle(notification.getTitle())
+              .setContentText(notification.getBody())
+              .setPriority(NotificationCompat.PRIORITY_DEFAULT)
+              .setContentIntent(pendingIntent)
+              .setAutoCancel(true);
+
+
+      String sound = notification.getSound();
+      if (sound != null) {
+        mBuilder.setSound(Uri.parse(sound));
+
+      }
+
+      JSONObject extra = notification.getExtra();
+      if (extra != null) {
+        Bundle extras = new Bundle();
+        extras.putString("data", extra.toString());
+        mBuilder.addExtras(extras);
+        intent.putExtras(extras);
+      }
+
+      // notificationId is a unique int for each notification that you must define
+      notificationManager.notify(Integer.valueOf(notification.getId()), mBuilder.build());
+      ids.put(notification.getId());
+    }
+    call.success(new JSObject().put("ids", ids));
+  }
+
+  @PluginMethod()
+  public void cancel(PluginCall call) {
+    List<JSONObject> notifications = null;
+    try {
+      notifications = call.getArray("notifications").toList();
+    } catch (JSONException e) {
+    }
+    if (notifications == null || notifications.size() == 0) {
+      call.error("Must provide notifications array as notifications option");
+      return;
+    }
+    NotificationManagerCompat notificationManager = NotificationManagerCompat.from(this.getContext());
+    for (JSONObject notificationToCancel : notifications) {
+      try {
+        Integer notificationId = notificationToCancel.getInt("id");
+        notificationManager.cancel(notificationId);
+      } catch (JSONException e) {
+        call.error("id field missing in Notification cancel method ");
+        return;
+      }
+    }
+    call.success();
+  }
+
+  @PluginMethod()
+  public void getPending(PluginCall call) {
+    // IOS specific
```

Comment:
We can probably keep track of the scheduled notifications and return that instead of asking the system.
Anyway it's ok if we don't have this yet, but I would rather return an error with non implemented message than returning success